File: projection_library_kernel_v1.1/projection_library/backend/api/schemas/validation.py
"""Pydantic schemas for the M5 historical validation endpoint."""

from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


Severity = Literal["block", "error", "warning", "info"]


class ValidationIssueDTO(BaseModel):
    rule_id: str
    severity: Severity
    voice_id: str | None = None
    year: str | None = None
    message: str
    context: dict = Field(default_factory=dict)


class ValidationSummary(BaseModel):
    block: int = 0
    error: int = 0
    warning: int = 0
    info: int = 0


class ValidationReportResponse(BaseModel):
    project_id: str
    phase: Literal["historical"] = "historical"
    summary: ValidationSummary
    issues: list[ValidationIssueDTO]
    can_proceed: bool
    computed_at: datetime


__all__ = [
    "Severity",
    "ValidationIssueDTO",
    "ValidationReportResponse",
    "ValidationSummary",
]

File: projection_library_kernel_v1.1/projection_library/frontend/src/types/validation.ts
export type Severity = 'block' | 'error' | 'warning' | 'info'

export interface ValidationIssue {
  rule_id: string
  severity: Severity
  voice_id: string | null
  year: string | null
  message: string
  context: Record<string, unknown>
}

export interface ValidationSummary {
  block: number
  error: number
  warning: number
  info: number
}

export interface HistoricalValidationReport {
  project_id: string
  phase: 'historical'
  summary: ValidationSummary
  issues: ValidationIssue[]
  can_proceed: boolean
  computed_at: string
}

export interface LatestValidationReport {
  project_id: string
  snapshot_id: string
  run_timestamp: string
  status: string
  summary: Partial<ValidationSummary> & Record<string, unknown>
  issues: ValidationIssue[]
}
